pallets/eth-bridge/src/lib.rs

```diff
@@ -514,22 +523,30 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             ensure_none(origin)?;
 
+            ensure!(events_fraction.is_valid(), Error::<T>::InvalidEventVote);
             // Validate that the range is the current one.
             let active_range =
                 Self::active_ethereum_range().ok_or(Error::<T>::EventScanningDisabled)?;
-            ensure!(range == active_range, Error::<T>::EventScanningDisabled);
+            ensure!(range == active_range.range, Error::<T>::NonActiveEthereumRange);
+            ensure!(
+                active_range.status == EventProcessingStatus::UnderValidation,
```

Comment:
```suggestion
                active_range.status == EventProcessingStatus::ValidationInProgress,
```
This might be a personal choice but when I read `UnderValidation` I thought it mean its below the required threshold :)